File: 0841-keys-and-rooms/python/bfs.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        visited = [False for _ in range(len(rooms))]

        queue = []
        queue.append(0)

        while len(queue) >  0:
            src = queue[0]
            queue = queue[1:]

            visited[src] = True

            for room in rooms[src]:
                if visited[room] == False:
                    queue.append(room)

        for v in visited:
            if v == False:
                return False

        return True

File: 0841-keys-and-rooms/golang/bfs.go
func canVisitAllRooms(rooms [][]int) bool {
    visited := make([]bool, len(rooms))
    for i := range visited {
        visited[i] = false
    }

    var queue[] int;
    queue = append(queue, 0)

    for len(queue) > 0 {
        src := queue[0]
        queue = queue[1:]

        visited[src] = true

        for _, room := range rooms[src] {
            if visited[room] == false {
                queue = append(queue, room)
            }
        }
    }

    for _, room := range visited {
        if room == false {
            return false
        }
    }

    return true

}